app/src/main/java/pm/gnosis/heimdall/ui/onboarding/RestoreAccountActivity.kt

```diff
@@ -50,12 +52,28 @@ class RestoreAccountActivity : BaseActivity() {
                             onNext = { layout_restore_account_mnemonic_input_layout.error = null },
                             onError = Timber::e)
 
-    private fun onMnemonicValidation(isValid: Boolean) {
-        if (isValid) {
-            layout_restore_account_mnemonic_input_layout.error = null
-            disposables += saveAccountWithMnemonicDisposable(layout_restore_account_mnemonic.text.toString())
-        } else {
-            layout_restore_account_mnemonic_input_layout.error = "Invalid mnemonic phrase"
+    private fun onMnemonicValidation(mnemonicResult: Bip39ValidationResult) {
+        when (mnemonicResult) {
+            is ValidMnemonic -> {
+                layout_restore_account_mnemonic_input_layout.error = null
+                disposables += saveAccountWithMnemonicDisposable(layout_restore_account_mnemonic.text.toString())
+            }
+            is InvalidEntropy -> {
+                layout_restore_account_mnemonic_input_layout.error = getString(R.string.invalid_mnemonic)
+            }
+            is InvalidChecksum -> {
+                layout_restore_account_mnemonic_input_layout.error = getString(R.string.invalid_mnemonic)
+            }
+            is MnemonicNotInWordlist -> {
+                val wordLists = BIP39_WORDLISTS.keys.joinToString()
```

Comment:
Having this in the view feels kind of weird, as it is a pretty specifix implementation detail